libraries/AP_Networking/AP_Networking.cpp

```diff
@@ -0,0 +1,393 @@
+
+#include "AP_Networking.h"
+
+#if AP_NETWORKING_ENABLED
+
+#include "AP_Math/definitions.h"
+#include <GCS_MAVLink/GCS.h>
+
+#if CONFIG_HAL_BOARD == HAL_BOARD_CHIBIOS
+    #include <hal_mii.h>
+    #include <lwip/sockets.h>
+#else
+    #include <arpa/inet.h>
+    #include <sys/socket.h>
+   
+#endif
+
+extern const AP_HAL::HAL& hal;
+
+const AP_Param::GroupInfo AP_Networking::var_info[] = {
+
+    // @Param: ENABLED
+    // @DisplayName: Enable Networking
+    // @Description: Enable Networking
+    // @Values: 0:Disable,1:Enable
+    // @RebootRequired: True
+    // @User: Advanced
+    AP_GROUPINFO_FLAGS("ENABLED",  0, AP_Networking, _param.enabled, 0, AP_PARAM_FLAG_ENABLE),
+
+    // @Param: IPADDR0
+    // @DisplayName: IPv4 Address MSB
+    // @Description: Allows setting static IP address. Example: 192.xxx.xxx.xxx
+    // @Range: 0 255
+    // @RebootRequired: True
+    // @User: Advanced
+    AP_GROUPINFO("IPADDR0", 1,  AP_Networking,    _param.ipaddr[0],   AP_NETWORKING_DEFAULT_STATIC_IP_ADDR0),
+
+    // @Param: IPADDR1
+    // @DisplayName: IPv4 Address 2nd byte
+    // @Description: Allows setting static IP address. Example: xxx.168.xxx.xxx
+    // @Range: 0 255
+    // @RebootRequired: True
+    // @User: Advanced
+    AP_GROUPINFO("IPADDR1", 2,  AP_Networking,    _param.ipaddr[1],   AP_NETWORKING_DEFAULT_STATIC_IP_ADDR1),
```

Comment:
What happened to the `4` and `6` for IPV4 vs IPV6 distinction?

Reply:
We're sticking with 4 only for now. Plus, @tridge has a new Param object he wants to implement for quad unit8 for IPs so this stuff will be changing in the future anyway. Also, it's common to have IP meaning v4 and ipv6 for 6 so I think this is fine for now anyway.